src/noise.jl

```diff
@@ -1,15 +1,34 @@
 abstract type AbstractNoise end
 
-struct StandardGaussian{T} <: AbstractNoise
+# Noise without an explicit type def
+## Given scaling matrix
+noise(s::AbstractMatrix, t) = s * randn(size(s, 1))
+## Given diagonal noise vector
+noise(s::AbstractVector, t) = s .* randn(length(s))
+
+struct Gaussian{T, Btype} <: AbstractNoise
     size::T
+    B::Btype # XXX: Probably should be the covariance matrix and not nothing, save on alloc
 end
 
-# Must include RNG support here
-noise(s::StandardGaussian{<:Integer}, t) = randn(s.size)
-noise(s::StandardGaussian{<:Tuple}, t) = randn(s.size...)
+StandardGaussian(size::T) where T = Gaussian{T, Nothing}(size, nothing)
+
+# TODO: Must include RNG support here
+noise(s::Gaussian{<:Integer, <:Nothing}, t) = randn(s.size)
+noise(s::Gaussian{<:Tuple, <:Nothing}, t) = randn(s.size...)
+
+# Scaling draws by a matrix
+noise(s::Gaussian{<:Integer, AbstractArray}, t) = S.B * randn(s.size)
+noise(s::Gaussian{<:Tuple, AbstractArray}, t) = s.B * randn(s.size...)
+
 
 struct DefinedNoise{T} <: AbstractNoise
     values::T
 end
 
 noise(s::DefinedNoise{<:Vector}, t) = s.values[t]
+
+# Likelihood definitions
+StatsBase.loglikelihood(x, noise::Gaussian{<:Any, <:Nothing}, t) = loglikelihood(MvNormal(diagm(ones(noise.size))), x)
+StatsBase.loglikelihood(x, noise::Gaussian, t) = loglikelihood(MvNormal(noise.B), x)
+StatsBase.loglikelihood(x, noise::AbstractMatrix, t) = loglikelihood(MvNormal(noise), x)
```

Comment:
Is there a reason this cannot or should not use the DiffEqNoiseProcess interface? It looks like the likelihoods would just need to be added on there.

Reply:
My two cents: if DiffEqNoiseProcess supports (i.i.d.) shock processes with a variety of distributions then we should use it.

Reply:
It has a few, and could use more. It defines it in the continuous context but those should be useable here? What kinds of shock distributions are necessary?

Reply:
For those noises, I think maybe Gaussian, Student-t, Gamma, Laplace... but the key is really just i.i.d. shocks over discrete time periods.

Reply:
Ahh some of those don't have a continuous representation, so let's keep this separate.